File: Mediator/Golang/LockingSystem.go
package main

import "fmt"

type LockingSystem struct {
	status bool
	command Command
}

func NewLockingSystem () *LockingSystem {
	return &LockingSystem{false, nil}
}

func (lockingSystem *LockingSystem) lock(command Command)  {
	lockingSystem.command = command
	lockingSystem.status = true
}

func (lockingSystem *LockingSystem) unlock()  {
	lockingSystem.status = false
}

func (lockingSystem *LockingSystem) checkStatus() bool {
	if (lockingSystem.status) {
        fmt.Println("The system currently used by:")
		lockingSystem.command.describe()
	} 

	return lockingSystem.status
}

File: Mediator/Javascript/LockingSystem.js
class LockingSystem {
  constructor() {
    this.status = false;
    this.command = null;
  }

  lock(command) {
    this.status = true;
    this.command = command;
  }

  unlock() {
    this.status = false;
    this.command = null;
  }

  checkStatus() {
    if (this.status) {
      console.log("The system is currently used by:")
      this.command.describe();
    }

    return this.status;
  }
}

export default LockingSystem;